mephisto/abstractions/providers/mturk/mturk_utils.py

```diff
@@ -738,3 +743,111 @@ def expire_and_dispose_hits(
+        # Too recent, no cleanup needed
+        return
+
+    print(
+        f"It's been more than a day since you last ran a job "
+        f"with {requester_name}, checking for outstanding tasks..."
+    )
+
+    requester = db.find_requesters(requester_name=requester_name)[0]
+    client = requester._get_client(requester._requester_name)
+
+    def hit_is_broken(hit: Dict[str, Any]) -> bool:
+        return (
+            hit["NumberOfAssignmentsCompleted"] == 0
+            and hit["HITStatus"] != "Reviewable"
+        )
+
+    query_time = time.time()
+    outstanding_hit_types = get_outstanding_hits(client)
+    if time.time() - query_time > 60:
+        print(
+            "That took a while! You may want to run `mephisto scripts "
+            "cleanup mturk` later to clear out some of the older HIT types.\n"
+        )
+    broken_hit_types = {
+        k: [h for h in v if hit_is_broken(h)]
+        for (k, v) in outstanding_hit_types.items()
+        if any([h for h in v if hit_is_broken(h)])
+    }
+    num_hit_types = len(broken_hit_types.keys())
+
+    sum_hits = sum([len(broken_hit_types[x]) for x in broken_hit_types.keys()])
+
+    last_cleanup_times[requester_name] = time.time()
+    with open(cleanups_path, "w+") as cleanups_file:
+        json.dump(last_cleanup_times, cleanups_file)
+
+    if sum_hits == 0:
+        print(f"No broken HITs detected. Continuing!")
+        return
+
+    print(
+        f"The requester {requester_name} has {num_hit_types} outstanding HIT "
+        f"types, with {sum_hits} suspected active or broken HITs.\n"
+        "This may include tasks that are still in-flight, but also "
+        "tasks have been improperly shut down and need cleanup.\n "
+        "Please review and dispose HITs below."
+    )
+
+    hits_to_dispose: Optional[List[Dict[str, Any]]] = []
+    confirm_string = "Enter anything to confirm removal of the following HITs:\n"
+    for hit_type in broken_hit_types.keys():
+        hit_count = len(broken_hit_types[hit_type])
+        cur_title = broken_hit_types[hit_type][0]["Title"]
+        creation_time = broken_hit_types[hit_type][0]["CreationTime"]
+        creation_time_str = creation_time.strftime("%m/%d/%Y, %H:%M:%S")
+        print(f"HIT TITLE: {cur_title}")
+        print(f"LAUNCH TIME: {creation_time_str}")
+        print(f"HIT COUNT: {hit_count}")
+        should_clear = ""
+        while not (should_clear.startswith("y") or should_clear.startswith("n")):
+            should_clear = input(
+                "Should we cleanup this hit type? (y)es or (n)o: " "\n>> "
+            ).lower()
+        if should_clear.startswith("y"):
+            hits_to_dispose += broken_hit_types[hit_type]
+            confirm_string += (
+                f"{hit_count} hits from {creation_time_str} for HIT Type: {cur_title}\n"
+            )
+
+    if len(hits_to_dispose) == 0:
+        print("No HITs selected for disposal. Continuing")
+        return
+
+    input(confirm_string)
```

Comment:
Shouldn't we get user confirmation here?

Reply:
I was a bit lazy here, using it as an opportunity to exit via `^C`. Let me put it in a `y`/`n` loop.